drivers/w1/w1_ds248x.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2022 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_W1_W1_DS248X_H_
+#define ZEPHYR_DRIVERS_W1_W1_DS248X_H_
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/zephyr.h>
+
+#define CMD_1WT	 0x78
+#define CMD_1WSB 0x87
+#define CMD_1WRB 0x96
+#define CMD_1WWB 0xa5
+#define CMD_1WRS 0xb4
+#define CMD_CHSL 0xc3 /* DS2482-800 only */
+#define CMD_ADJP 0xc3 /* DS2484 only */
+#define CMD_WCFG 0xd2
+#define CMD_SRP	 0xe1
+#define CMD_DRST 0xf0
+
+#define REG_NONE    0x00 /* special value */
+#define REG_CONFIG  0xc3
+#define REG_DATA    0xe1
+#define REG_STATUS  0xf0
+#define REG_CHANNEL 0xd2 /* DS2482-800 only */
+#define REG_PORT    0xb4 /* DS2484 only */
+
+/*
+ * Device Configuration Register
+ */
+#define DEVICE_APU_pos 0
+#define DEVICE_APU_msk BIT(0)
+#define DEVICE_PDN_pos 1      /* DS2484 only */
+#define DEVICE_PDN_msk BIT(1) /* DS2484 only */
+#define DEVICE_SPU_pos 2
+#define DEVICE_SPU_msk BIT(2)
+#define DEVICE_1WS_pos 3
+#define DEVICE_1WS_msk BIT(3)
+
+/*
+ * Status Register
+ */
+#define STATUS_1WB_pos 0
+#define STATUS_1WB_msk BIT(0)
+#define STATUS_PPD_pos 1
+#define STATUS_PPD_msk BIT(1)
+#define STATUS_SD_pos  2
+#define STATUS_SD_msk  BIT(2)
+#define STATUS_LL_pos  3
+#define STATUS_LL_msk  BIT(3)
+#define STATUS_RST_pos 4
+#define STATUS_RST_msk BIT(4)
+#define STATUS_SBR_pos 5
+#define STATUS_SBR_msk BIT(5)
+#define STATUS_TSB_pos 6
+#define STATUS_TSB_msk BIT(6)
+#define STATUS_DIR_pos 7
+#define STATUS_DIR_msk BIT(7)
+
+/*
+ * Channel Selection Codes, DS2482-800 only
+ */
+#define CHSL_IO0 0xf0
+#define CHSL_IO1 0xe1
+#define CHSL_IO2 0xd2
+#define CHSL_IO3 0xc3
+#define CHSL_IO4 0xb4
+#define CHSL_IO5 0xa5
+#define CHSL_IO6 0x96
+#define CHSL_IO7 0x87
+
+/*
+ * Channel Selection Codes (read back values), DS2482-800 only
+ */
+#define CHSL_IO0_RB 0xb8
+#define CHSL_IO1_RB 0xb1
+#define CHSL_IO2_RB 0xaa
+#define CHSL_IO3_RB 0xa3
+#define CHSL_IO4_RB 0x9c
+#define CHSL_IO5_RB 0x95
+#define CHSL_IO6_RB 0x8e
+#define CHSL_IO7_RB 0x87
+
+/*
+ * Port Configuration Register, DS2484 only
+ */
+#define PORT_VAL0_pos 0
+#define PORT_VAL0_msk BIT(0)
+#define PORT_VAL1_pos 1
+#define PORT_VAL1_msk BIT(1)
+#define PORT_VAL2_pos 2
+#define PORT_VAL2_msk BIT(2)
+#define PORT_VAL3_pos 3
+#define PORT_VAL3_msk BIT(3)
+
+/*
+ * Bit Byte
+ */
+#define BIT_CLR_msk 0
+#define BIT_SET_msk BIT(7)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int ds248x_reset_bus(const struct i2c_dt_spec *spec);
```

Comment:
I'd probably document these, but since this is a private header, I'm not sure if it is required

Reply:
True. Not sure what the usual way is, though.

Reply:
I think it was about the `extern "C"`-declaration. But I agree, this shouldn't be necessary.